tests/test_masks.py

```diff
@@ -0,0 +1,453 @@
+    assert (bitmap_masks[0].masks == raw_masks[0]).all()
+    assert (bitmap_masks[range(2)].masks == raw_masks[range(2)]).all()
+
+
+def test_polygon_mask_init():
+    # init with empty masks
+    raw_masks = []
+    polygon_masks = BitMapMasks(raw_masks, 28, 28)
+    assert len(polygon_masks) == 0
+    assert polygon_masks.height == 28
+    assert polygon_masks.width == 28
+
+    # init with masks contain 3 instances
+    raw_masks = dummy_raw_polygon_masks((3, 28, 28))
+    polygon_masks = PolygonMasks(raw_masks, 28, 28)
+    assert isinstance(polygon_masks.masks, list)
+    assert isinstance(polygon_masks.masks[0], list)
+    assert isinstance(polygon_masks.masks[0][0], np.ndarray)
+    assert len(polygon_masks) == 3
+    assert polygon_masks.height == 28
+    assert polygon_masks.width == 28
+
+    # init with raw masks of unsupported type
+    with pytest.raises(AssertionError):
+        raw_masks = [[[]]]
+        PolygonMasks(raw_masks, 28, 28)
+
+        raw_masks = [dummy_raw_polygon_masks((3, 28, 28))]
+        PolygonMasks(raw_masks, 28, 28)
+
+
+def test_polygon_mask_rescale():
+    # rescale with empty polygon masks
+    raw_masks = dummy_raw_polygon_masks((0, 28, 28))
+    polygon_masks = PolygonMasks(raw_masks, 28, 28)
+    rescaled_masks = polygon_masks.rescale((56, 72))
+    assert rescaled_masks.height == 56
+    assert rescaled_masks.width == 56
+
+    # rescale with polygon masks contain 3 instances
+    raw_masks = dummy_raw_polygon_masks((3, 28, 28))
+    polygon_masks = PolygonMasks(raw_masks, 28, 28)
+    rescaled_masks = polygon_masks.rescale((56, 72))
+    assert rescaled_masks.height == 56
+    assert rescaled_masks.width == 56
+
+
+def test_polygon_mask_resize():
+    # resize with empty polygon masks
+    raw_masks = dummy_raw_polygon_masks((0, 28, 28))
+    polygon_masks = PolygonMasks(raw_masks, 28, 28)
+    rescaled_masks = polygon_masks.resize((56, 72))
+    assert rescaled_masks.height == 56
+    assert rescaled_masks.width == 72
+
+    # resize with polygon masks contain 3 instances
+    raw_masks = dummy_raw_polygon_masks((3, 28, 28))
+    polygon_masks = PolygonMasks(raw_masks, 28, 28)
+    rescaled_masks = polygon_masks.resize((56, 72))
+    assert rescaled_masks.height == 56
+    assert rescaled_masks.width == 72
+
+
+def test_polygon_mask_flip():
+    # flip with empty polygon masks
+    raw_masks = dummy_raw_polygon_masks((0, 28, 28))
+    polygon_masks = PolygonMasks(raw_masks, 28, 28)
+    flipped_masks = polygon_masks.flip(flip_direction='horizontal')
+    assert len(flipped_masks) == 0
+
+    # horizontally flip with polygon masks contain 3 instances
+    raw_masks = dummy_raw_polygon_masks((3, 28, 28))
+    polygon_masks = PolygonMasks(raw_masks, 28, 28)
+    flipped_masks = polygon_masks.flip(flip_direction='horizontal')
+    flipped_flipped_masks = flipped_masks.flip(flip_direction='horizontal')
+    assert (polygon_masks.to_ndarray() == flipped_flipped_masks.to_ndarray()
+            ).all()
```

Comment:
This is not safe. A better way is to check the flipped masks directly. `assert polygon_masks.to_ndarray() == flipped_masks.to_ndarray()[:, :, ::-1]`